src/main/java/group1/mips_simulator/components/instructionExecution/InstructionExecutions.java

```diff
@@ -545,4 +546,71 @@ public ExecutionResult execute_sir(Computer computer, RXIA_Instruction i) {
         targetReg.write(newValue);
         return new ExecutionResult(computer.currentPcPlus1());
     }
+
+    /**
+     * SRC r, count, L/R, A/L
+     * 30(octal)
+     * Shift Register by Count
+     * c(r) is shifted left (L/R =1) or right (L/R = 0) either logically (A/L = 1) or arithmetically (A/L = 0)
+     * XX, XXX are ignored
+     * Count = 0…15
+     * If Count = 0, no shift occurs
+     */
+    public ExecutionResult execute_src(Computer computer, Bitwise_Instruction i) {
+        Register targetReg = computer.cpu.regfile.getGPR(i.getR().value);
+
+        boolean shiftLeft = (i.getLeftRight().value == 1);
```

Comment:
what is this function, `getLeftRight` ?
most of the other ones i kinda figure out from context clues, but i missed this one

Reply:
Are we shifting left or right. It's simply called `L/R` in the documentation which is kinda confusing :/ 

If you have a better name suggestion I'm all ears